frame/vesting/src/lib.rs

```diff
@@ -61,41 +60,139 @@ use frame_support::traits::{
 use frame_system::{ensure_signed, ensure_root, pallet_prelude::*};
 pub use weights::WeightInfo;
 pub use pallet::*;
+pub use vesting_info::*;
 
 type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
 type MaxLocksOf<T> = <<T as Config>::Currency as LockableCurrency<<T as frame_system::Config>::AccountId>>::MaxLocks;
 
 const VESTING_ID: LockIdentifier = *b"vesting ";
 
-/// Struct to encode the vesting schedule of an individual account.
-#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug)]
-pub struct VestingInfo<Balance, BlockNumber> {
-	/// Locked amount at genesis.
-	pub locked: Balance,
-	/// Amount that gets unlocked every block after `starting_block`.
-	pub per_block: Balance,
-	/// Starting block for unlocking(vesting).
-	pub starting_block: BlockNumber,
+// Module to enforce private fields on `VestingInfo`
+mod vesting_info {
```

Comment:
IMO we can improve code readability by having more, shorter files. This PR adds ~1000 lines to `lib.rs` anyway, and it's not clear how much of it could potentially be externalized. However, as this is already factored into a submodule, we might as well make it into its own file.

Reply:
i agree. i will pull this out once i have addressed the other comments from today

Reply:
9924ddd